hdfs3/core.py

```diff
@@ -315,12 +315,16 @@ def __del__(self):
 
     def mkdir(self, path):
         out = _lib.hdfsCreateDirectory(self._handle, ensure_byte(path))
-        return out == 0
+        if out != 0:
+            raise IOError('Create directory failed')
```

Comment:
In the future we should probably also test these errors as we write code for them.  I can handle this in the next sweep though.